infer/src/pulse/PulseSummary.ml

```diff
@@ -87,8 +88,17 @@ let exec_summary_of_post_common ({InterproceduralAnalysis.proc_desc} as analysis
       AbductiveDomain.Summary.of_post (Procdesc.get_attributes proc_desc) location astate
     in
     match (summary_result : _ result) with
-    | Ok summary ->
-        exec_domain_of_summary summary
+    | Ok summary -> (
+        let r = exec_domain_of_summary summary in
+        match r with
+        | InfiniteProgram _ ->
+            let curnode = Metadata.get_infinite_node in
+            let curloc = Procdesc.Node.get_loc (curnode ()) in
```

Comment:
I think you could store the current node inside `InfiniteProgram` at the point it is created and then read it back here instead of maintaining a ref inside `AbstractInterpreter.DisjunctiveMetadata`. You can get the node under analysis at any point in pulse with `AnalysisState.get_node_exn`.